docs/pywbemcligeneraloptions.rst

```diff
@@ -61,166 +70,470 @@ command, general the options redefinition's are not retained between command
-    operations.
-
-* **--pull-max-cnt** -  controls the CIM-XML open and pull request parameter
-  ``MaxObjectCount`` that tells the server the maximum number of objects
-  to be returned for each pull request if pull operations are used. This must
-  be  a positive non-zero integer. The default is 1000. See :ref:`Pywbemcli and the
-  DMTF pull operations` for more information on pull operations.
-* **--mock-server/-m** - Defines one or more files that define a mock server that
-  can be used to define a mock WBEM server in the pywbemcli process so that
-  pywbemcli commands without access to a real server. When this option is used
-  to define a WBEM server the security options (ex. ``--user``) are irrevalent;
-  they may be included but are not used.
-
-  The following example creates a mock server with two files defining the
-  mock data, shows what parameters are defined for the connection, and then
-  saves that connection named ``mymockserver``:
 
-  .. code-block:: text
+.. _`--user general option`:
+
+--user general option
+^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--user/-u`` is a string that is the user name on the WBEM
+server if a user name is required by the WBEM server to authenticate the client.
+
+.. _`--password general option`:
+
+--password general option
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--password/-p`` general option is a string that is the
+password for the WBEM server. This option is normally required if the
+:ref:`--user general option` is defined.  If passwords are saved into the
+:term:`connections file` they are not encrypted in the file.
+
+If the WBEM operations performed by any pywbemcli command require a password,
+the password is prompted for if the ``--user general option`` is set (in both
+modes of operation) and the ``--password general option`` is not set.
+
+.. code-block:: text
+
+      $ pywbemcli -s http://localhost -d root/cimv2 -u user class get
+      Enter password: <password>
+      . . . <The display output from get class>
+
+If both the ``--user`` and ``--password`` options are set, the command is
+executed without a password prompt:
+
+.. code-block:: text
+
+      $ pywbemcli -s http://localhost -d root/cimv2 -u user -p blah class get
+      . . . <The display output from get class>
+
+If the operations performed by a particular pywbemcli command do not
+require a password or no user is supplied, no password is prompted for example:
+
+.. code-block:: text
+
+      $ pywbemcli --help
+      . . . <help output>
+
+For script integration, it is important to have a way to avoid the interactive
+password prompt. This can be done by storing the password string in an
+environment variable or specifying it on the command line.
+See :ref:`Environment variables for general options`
+
+The pywbemcli ``connection export`` command that outputs the (bash/windows)
+shell commands to set all needed environment variables:
```

Comment:
The Windows commands are different, so I suggest to show them in a separate example.

Reply:
Change to note that this is a linux type os example